### ISSUE TITLE
Resolved errors persists if there are any other errors in workspace

### ISSUE THREAD

**Issue by peku33** [microsoft/vscode-mypy#359]
## Diagnostic Data
- Python version (& distribution if applicable, e.g., Anaconda): `3.11.9`
- Type of virtual environment used (e.g., conda, venv, virtualenv, etc.): poetry managed .venv, with mypy locally installed and `"mypy-type-checker.importStrategy": "fromEnvironment"`
- Operating system (and version): tested on windows 11 x64 and mac os x
- Version of tool extension you are using: `2025.3.10801012`

### Expected Behavior
When there are two files with reported errors, when errors in one are fixed and file is saved, they should disappear from problems tab.

### Actual Behavior
If there is more then one files with issues - fixed errors in other files aren't removed.

## Reproduction Steps:
Create two files: `foo1.py` and `foo2.py` with same contents:

```python
def function1(_variable: str) -> None:
    pass


function1("aaa")
```

Save files, no errors should be reported (ok).

Introduce typing problem in both files, by replacing function call with:

```python
function1(123)
```

After saving errors in both files are being reported (ok).

Fix error in one of the file by restoring it to previous version and save it.

Reported error persists even though it has been resolved (problem).

## Logs:
Except for expected tracing logs, there is nothing worth mentioning. After saving fixed file mypy is being run again and only second file is reported as problematic to output, but problems from first file are not removed.

## Outcome When Attempting Debugging Steps:
Command line mypy works as expected - it reports problematic code only.

## Extra Details

I use following config settings for the tool:
```
    "mypy-type-checker.importStrategy": "fromEnvironment",
    "mypy-type-checker.reportingScope": "workspace",
```
Mypy in environment is `mypy 1.15.0 (compiled: yes)`

I've encountered this error in different projects on different machines, so I believe it's platform and structure independent. It's classic, non-multiroot workspace.

**Comment by karthiknadig** [comment]
The bug seems to be in this function:
https://github.com/microsoft/vscode-mypy/blob/78868dc55ef588256b03b2043499c4f5ca25dcea/bundled/tool/lsp_server.py#L204

We keep track of files where reports are sent out, but we are not clearing it out in the follow up run.

**Comment by karthiknadig** [comment]
We are open to taking a PR for fixing this issue.

**Comment by peku33** [comment]
This must have been introduced in some of recent versions - I don't remember it happening before march 2025.

**Comment by tdscheper** [comment]
The `==` here should be `in`
https://github.com/microsoft/vscode-mypy/blob/78868dc55ef588256b03b2043499c4f5ca25dcea/bundled/tool/lsp_server.py#L263-L267
This bug was introduced last month with commit 75590161a8b1503d2f4ce2cb6b9c4930d4be07f4, see the [git blame](https://github.com/microsoft/vscode-mypy/blame/78868dc55ef588256b03b2043499c4f5ca25dcea/bundled/tool/lsp_server.py#L263).
Used to be `if reportingScope == "workspace"`, now is `if reportingScope == ("workspace", "custom")`, should be `if reportingScope in ("workspace", "custom")`

**Comment by karthiknadig** [comment]
@tdscheper Thanks for investigating. created a PR with the fix https://github.com/microsoft/vscode-mypy/pull/361
